sklearn/discriminant_analysis.py

```diff
@@ -254,6 +255,22 @@ def __init__(self, solver='svd', shrinkage=None, priors=None,
         self.store_covariance = store_covariance  # used only in svd solver
         self.tol = tol  # used only in svd solver
 
+    @property
+    def covariance_(self):
+        if hasattr(self, '_covariance_') and not self.store_covariance:
+            warnings.warn("from version 0.21 'covariance_' will be stored "
+                          "only if 'store_covariance' is True",
+                          FutureWarning)
+        return self._covariance_
+
+    @covariance_.setter
```

Comment:
(I'm also okay with just removing the setter altogether, as we have no reason to expect users to set it.)